dissect/target/plugins/os/windows/notifications.py

```diff
@@ -1,15 +1,150 @@
+  QWORD    LastNotificationDate;
+  DWORD    NextNotificationId;
+  DWORD    Unknown;
+  char     Padding[8];
+} ChunkHeader;                    // size: 0x20
+
+typedef struct
+{
+  WORD InUse;           // ??
+  WORD NextTileWrite;   // ??
+  WORD NextToastWrite;  // ??
+  BYTE Flags[2];
+} ChunkInfo;
+
+typedef struct
+{
+  QWORD    Timestamp1;              // ??
+  QWORD    Timestamp2;              // Is this time to refresh?
+  char     Uri[1024];               // Is this the correct size?
+  char     Padding[0x818 - 0x410];
+} PushDescriptor;                   // size: 0x818
+
+typedef struct
+{
+  DWORD    Id;
+  DWORD    Zero;              // ??
+  QWORD    Timestamp;
+  WORD     Unknown;
+  WORD     DataLength;
+  char     Data[DataLength];
+  char     Padding[0x118 - 0x14 - DataLength];
+} BadgeContent;  // size: 0x118
+
+typedef struct
+{
+  DWORD      UniqueId;        // ??
+  DWORD      Zero;
+  QWORD      TS_Expiry;
+  QWORD      TS;
+  BYTE       Type;            // ??
+  BYTE       Index;
+  WORD       ContentLength;
+  wchar_t    Name[18];
+} TileDescriptor;             // size: 0x40
+
+typedef struct
+{
+  DWORD           UniqueId;        // ??
+  DWORD           Zero;
+  QWORD           TS_Expiry;
+  QWORD           TS;
+  BYTE            Type;           // ??
+  BYTE            Index;
+  WORD            ContentLength;
+  wchar_t         Name1[17];
+  wchar_t         Name2[17];
+} ToastDescriptor;  // size: 0x60
+
+typedef struct
+{
+  char Content[0x1400];
+} DataXML;  // 0x1400
+
+typedef struct
+{
+  ChunkHeader     Header;            // Only populated for first chunk, else zeroed
+  ChunkInfo       Info;
+  PushDescriptor  Push;
+  BadgeContent    BadgeXml;
+  TileDescriptor  Tiles[5];          // start @ 0x958
+  DataXML         TileXml[5];
+
+  // For the in use chunks, 0x14 ToastDiscriptors have an Index, but there
+  // is space for more. Maybe this is used in case of deleted entries?
+  ToastDescriptor Toasts[0x14];      // start @ 0x6e98
+  char            Padding1[0x1e00];  // start @ 0x7618
+  DataXML         ToastXml[0x14];    // start @ 0x9418
+  char            Padding2[0x13f8];  // start @ 0x22418
+} Chunk;                             // size: 0x23810
+"""
+
+c_appdb = cstruct(endian="<")
+c_appdb.load(appdb_def)
+
+APPDB_MAGIC = b"DNPW"
+NUM_APPDB_CHUNKS = 256
+
+AppDBChunkRecord = create_extended_descriptor([UserRecordDescriptorExtension])(
+    "windows/notification/appdb_chunk",
+    [
+        ("datetime", "last_notification_date"),
+        ("varint", "next_notification_id"),
+        ("datetime", "push_ts1"),
+        ("datetime", "push_ts2"),
+        ("uri", "push_uri"),
+        ("varint", "badge_id"),
+        ("datetime", "badge_ts"),
+        ("string", "badge_data"),
+    ],
+)
+
+AppDBTileRecord = create_extended_descriptor([UserRecordDescriptorExtension])(
+    "windows/notification/appdb_tile",
+    [
+        ("varint", "id"),
+        ("datetime", "ts"),
```

Comment:
Does it make sense to swap these so that the `ts` is first? Good for sorting.

Reply:
These are part of a grouped record so I don't know if sorting makes sense here.